### DISCUSSION TITLE
package of wheel built using python3.8 on windows cannot be imported

### DISCUSSION THREAD

**Discussion by zoj613** [pypa/cibuildwheel#1094] [Q&A]
### Description

I'm trying to build wheels for a package and thee CI fails when trying to for python 3.8, but works fine for python 3.7 and other platforms. I get a `ImportError: DLL load failed while importing _polyagamma: The specified module could not be found.` error.

### Build log

https://github.com/zoj613/polyagamma/runs/5841514828?check_suite_focus=true#step%3A7%3A318=

### CI config

https://github.com/zoj613/polyagamma/actions/runs/2098938033/workflow

**Comment by joerick** [comment]
This is more of a Q&A than an issue. Moving to Discussions.

**Reply by zoj613** [reply]
Fair enough. Do you have a suggestion for getting around this problem?

**Reply by Czaki** [reply]
disable testing, change the configuration to build only py3.8 wheel (for speedup), and download produced wheel.  Check its contents if `_polyagamma` is present in the wheel. 

This specific error means that `_polyagamma` may be missed or some file required by DLL could be missed. 

These are general steps to solve this problem. 

In your specific case the problem could be this step:
```
      - name: Install Dependencies and Cythonize extension
        run: |
          python3 -m pip install --upgrade pip setuptools wheel poetry
          poetry install --no-root
          poetry run cythonize polyagamma/*.pyx
 ```
You perform cythonization on host machine python using python 3.7. I do not know what `poetry cythonize` does but it may be a problem. Maybe you need to move this steep to CIBW_BEFORE_BUILD. 

But in general, I suggest using setuptools-based configuration for packages with binary extensions. Or another configuration that requires only `pip install path/to/package` to get the working package. 

Some time ago I got such link as suggestion how to configure poetry to work with binary extensions: https://github.com/davidcortesortuno/poetry_cython_proj
